### PR TITLE
Fixes #10379: When upgrading to 4.1, rudder.community.checkpromises.command=/bin/true option is lost 

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/10379

### CODE REVIEW

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 1]
```diff
@@ -464,6 +464,61 @@ find %{ruddervardir}/configuration-repository/.git %{ruddervardir}/configuration
...
 
+RUDDER_WEB_PROPERTIES="/opt/rudder/etc/rudder-web.properties"
+# Replace checkpromises with a hook
+HOOK_NAME="/opt/rudder/etc/hooks.d/policy-generation-node-ready/10-cf-promise-check"
+MIGRATED_HOOK_NAME="/opt/rudder/etc/hooks.d/policy-generation-node-ready/20-migrated-posthook"
+CURRENT_CHECKPROMISES=$(sed -n '/^rudder.community.checkpromises.command/s/rudder.community.checkpromises.command=//p' "${RUDDER_WEB_PROPERTIES}")
+[ "${CURRENT_CHECKPROMISES}" = "" ] && CURRENT_CHECKPROMISES=$(sed -n '/^rudder.nova.checkpromises.command/s/rudder.nova.checkpromises.command=//p' "${RUDDER_WEB_PROPERTIES}")
+if [ "${CURRENT_CHECKPROMISES}" = "/bin/true" ]
+then
+  # if /bin/true, just remove the hook
```
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: Shouldn't we only disable it as said in the ticket?

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 2]
```diff
@@ -464,6 +464,61 @@ find %{ruddervardir}/configuration-repository/.git %{ruddervardir}/configuration
...
+  # if default value, do nothing
+  true
+else
+  # if anything else present, put it in a hook replacing the distributed one
+  rm "${HOOK_NAME}"
+  cat > "${MIGRATED_HOOK_NAME}" << EOF
+#!/bin/sh
+
+# This file has been created by Rudder postinstall from your pre 4.1 rudder-web.properties file
+# The mathing property has been commented out
```
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: matching

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 3]
```diff
@@ -464,6 +464,61 @@ find %{ruddervardir}/configuration-repository/.git %{ruddervardir}/configuration
 ## Add execution permission for ncf-api on pre/post-hooks
 chmod -R 2750 %{ruddervardir}/configuration-repository/ncf/ncf-hooks.d/
 
+RUDDER_WEB_PROPERTIES="/opt/rudder/etc/rudder-web.properties"
+# Replace checkpromises with a hook
+HOOK_NAME="/opt/rudder/etc/hooks.d/policy-generation-node-ready/10-cf-promise-check"
+MIGRATED_HOOK_NAME="/opt/rudder/etc/hooks.d/policy-generation-node-ready/20-migrated-posthook"
+CURRENT_CHECKPROMISES=$(sed -n '/^rudder.community.checkpromises.command/s/rudder.community.checkpromises.command=//p' "${RUDDER_WEB_PROPERTIES}")
```
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: Shouldn't we trim this string just in case?

File: rudder-webapp/SPECS/rudder-webapp.spec (hunk 1 above)
Comment (peckpeck) [inline rudder-webapp/SPECS/rudder-webapp.spec]: The only way to currently disable a hook is to remove it

File: rudder-webapp/SPECS/rudder-webapp.spec (hunk 3 above)
Comment (peckpeck) [inline rudder-webapp/SPECS/rudder-webapp.spec]: java doesn't trim so the value should already be good

**Comment by peckpeck** [comment]
Commit modified

File: rudder-webapp/SPECS/rudder-webapp.spec (hunk 3 above)
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: 👍

File: rudder-webapp/SPECS/rudder-webapp.spec (hunk 1 above)
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: We should now rename it to .disabled

**Comment by peckpeck** [comment]
Commit modified

File: rudder-webapp/SPECS/rudder-webapp.spec [hunk 4]
```diff
@@ -464,6 +464,61 @@ find %{ruddervardir}/configuration-repository/.git %{ruddervardir}/configuration
...
+[ "${CURRENT_CHECKPROMISES}" = "" ] && CURRENT_CHECKPROMISES=$(sed -n '/^rudder.nova.checkpromises.command/s/rudder.nova.checkpromises.command=//p' "${RUDDER_WEB_PROPERTIES}")
+if [ "${CURRENT_CHECKPROMISES}" = "/bin/true" ]
+then
+  # if /bin/true, just remove the hook
+  rm "${HOOK_NAME}" "${HOOK_NAME}.disabled"
```
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: mv

File: rudder-webapp/SPECS/rudder-webapp.spec (hunk 1 above)
Comment (amousset) [inline rudder-webapp/SPECS/rudder-webapp.spec]: disable

**Comment by peckpeck** [comment]
Commit modified

**Pull request merged**